### ISSUE TITLE
Unexpected behaviour in google_container_cluster when using separate node pool and tags

### ISSUE THREAD

**Issue by 7adietri** [terraform-providers/terraform-provider-google#6930]
When running a second `terraform apply` for a `google_container_cluster` that is using `node_config.tags` on either the default node pool or a separate pool (but not both), Terraform wants to replace the cluster unexpectedly to "update" the tags. Our setup deletes the default node pool and uses just a separate one.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.12.29
+ provider.google v3.32.0
+ provider.google-beta v3.32.0

### Affected Resource(s)

* google_container_cluster

### Terraform Configuration Files

Example where the separate node pool uses tags, but the default node pool doesn't:

```tf
resource "google_container_cluster" "default" {
  provider = google-beta
  name     = var.cluster_name
  location = var.cluster_location

  release_channel {
    channel = var.release_channel
  }
  min_master_version = var.min_master_version

  remove_default_node_pool = true
  initial_node_count       = 1
  node_config {
    shielded_instance_config {
      enable_secure_boot = true
    }
  }

  private_cluster_config {
    enable_private_nodes    = true
    enable_private_endpoint = false
    master_ipv4_cidr_block  = var.master_ipv4_cidr_block

    master_global_access_config {
      enabled = true
    }
  }

  network         = google_compute_network.default.id
  networking_mode = "VPC_NATIVE"

  ip_allocation_policy {}

  enable_shielded_nodes = true

  workload_identity_config {
    identity_namespace = "${var.project}.svc.id.goog"
  }

  master_auth {
    username = ""
    password = ""

    client_certificate_config {
      issue_client_certificate = false
    }
  }
}

resource "google_container_node_pool" "default" {
  provider = google-beta
  name     = "default"
  cluster  = google_container_cluster.default.name
  location = var.cluster_location

  node_count = var.node_count

  node_config {
    machine_type = var.machine_type
    image_type   = "COS_CONTAINERD"
    tags         = ["gke-${var.cluster_name}-node"]

    shielded_instance_config {
      enable_secure_boot = true
    }

    metadata = {
      disable-legacy-endpoints = "true"
    }

    service_account = var.service_account

    oauth_scopes = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
    ]
  }
}
```

### Expected Behavior

The second `terraform apply` should run without any changes to resources.

#### Tags only in default node pool

The second `terraform apply` wants to replace the cluster to add the tags:

```
          ~ tags              = [ # forces replacement
              + "gke-test-node",
            ]
```

#### Tags only in separate node pool

The second `terraform apply` wants to replace the cluster to remove the tags:

```
          - tags              = [
              - "gke-test-node",
            ] -> null # forces replacement
```

### Steps to Reproduce

1. `terraform apply` (creates all resources as configured)
1. `terraform apply` (wants to replace the cluster unexpectedly)

### Important Factoids

Using the same tags on both the default node pool and the separate one seems to prevent this behaviour.

**Comment by ndmckinley** [comment]
Yep, I think this is a problem on the GKE side.  @venkykuberan can you file a bug with them asking them if they are seeing that tags which are applied to a node pool are returned from the API as if they are also applied to the cluster?  That's my read of the logs I have seen.

**Comment by ndmckinley** [comment]
Let me see if there's a quick workaround, though...

**Comment by ndmckinley** [comment]
I can't see one, so in the meantime I recommend adding

```
lifecycle {
  ignore_changes = ["tags"]
}
```

to your cluster.  Apologies for the inconvenience, we'll get this sorted out and come back with a better solution.  :)

**Comment by 7adietri** [comment]
> I can't see one, so in the meantime I recommend adding
> 
> ```
> lifecycle {
>   ignore_changes = ["tags"]
> }
> ```
> 
> to your cluster. Apologies for the inconvenience, we'll get this sorted out and come back with a better solution. :)

Thank you! I found that using the same tags on both the default node pool and the separate one seems to prevent this behaviour.

**Comment by ndmckinley** [comment]
Ah, great, yes!  Glad you found a workaround.  If you are okay with the tags being applied to both the node pool and the cluster, then adding them to both resources should stop the issue.  :)

**Comment by romankor** [comment]
+1

**Comment by tony-kerz** [comment]
i'm witnessing the same with the `preemptible` attribute,
but i think applying the same to node-pool and cluster as with `tags` is a similar workaround...
